src/main/java/org/mifos/pheebillpay/api/implementation/BillPaymentsController.java

```diff
@@ -22,8 +22,13 @@ public ResponseEntity<BillInquiryResponseDTO> billPayments(String tenantId, Stri
         BillInquiryResponseDTO billInquiryResponseDTO = new BillInquiryResponseDTO();
         String response = checkforValidity(tenantId, correlationId, callbackURL, payerFspId, body);
         if (!response.equals("Valid")) {
-            billInquiryResponseDTO.setTransactionId("Invalid Request: Mandatory Fields Missing, Missing field is " + response);
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(billInquiryResponseDTO);
+            if (response.equals("billId")) {
+                billInquiryResponseDTO.setTransactionId("Invalid Request: Bill Id Empty");
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body(billInquiryResponseDTO);
+            } else {
+                billInquiryResponseDTO.setTransactionId("Invalid Request: Mandatory Fields Missing, Missing field is " + response);
```

Comment:
why this error message is set to transaction Id?
The variable looks wrong

Reply:
okay changing